fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java

```diff
@@ -1725,20 +1740,31 @@ TExecBatchPlanFragmentsParams toThriftInBatch(
                 if (!inFlightInstanceIds.contains(instanceExecParam.instanceId)) {
                     continue;
                 }
-                int curTabletSinkDop = 0;
+                int curTableSinkDop = 0;
                 if (forceSetTableSinkDop) {
-                    curTabletSinkDop = fragment.getPipelineDop();
+                    DataSink dataSink = fragment.getSink();
+                    int dop = fragment.getPipelineDop();
+                    if (dataSink instanceof OlapTableSink) {
+                        curTableSinkDop = dop;
+                    } else if (dataSink instanceof IcebergTableSink) {
```

Comment:
Is it possible that `forceSetTableSinkDop` is true but `dataSink` isn't `OlapTableSink` or `IcebergTableSink`?

Reply:
this variable affects only `pipeline_sink_dop` currently.  and the `pipeline_sink_dop` is only used for olap and iceberg table in fragment_executor.cpp.